schemas/schema-core/src/main/java/org/fao/geonet/kernel/schema/subtemplate/SubtemplatesByLocalXLinksReplacer.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2001-2016 Food and Agriculture Organization of the
+ * United Nations (FAO-UN), United Nations World Food Programme (WFP)
+ * and United Nations Environment Programme (UNEP)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or (at
+ * your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ * Contact: Jeroen Ticheler - FAO - Viale delle Terme di Caracalla 2,
+ * Rome - Italy. email: geonetwork@osgeo.org
+ */
+
+package org.fao.geonet.kernel.schema.subtemplate;
+
+import org.apache.lucene.index.IndexReader;
+import org.jdom.Element;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SubtemplatesByLocalXLinksReplacer  {
+
+    private Map<String, AbstractReplacer> replacersDict = new HashMap<>();
+    private SearchManagerProxy searchManagerProxy;
+
+    public SubtemplatesByLocalXLinksReplacer(SearchManagerProxy searchManagerProxy) {
+        this.searchManagerProxy = searchManagerProxy;
+    }
+
+    public static String FORMAT = "format";
+    public static String CONTACT = "contact";
+    public static String EXTENT = "extent";
+
+    public Element replaceSubtemplatesByLocalXLinks(Element dataXml,
+                                                    String language,
+                                                    String templatesToOperateOn) {
+        IndexReader indexReader;
+        try {
+            indexReader = searchManagerProxy.getIndexReader();
+        } catch (IOException e) {
+            throw new RuntimeException("IOException, SubtemplatesByLocalXLinksReplacer, unable to init index reader.");
+        }
+        String localXlinkUrlPrefix = "local://" + language + "/subtemplate?uuid=";
```

Comment:
Seems that I set xlink the 'old' way, I should rather set it something like "local://srv/api/registries/entries/", beg for comments.